bridge_common/validate_job_api.py

```diff
@@ -0,0 +1,243 @@
+#
+# Refer to the README and COPYING files for full details of the license
+
+# Usage: opr = SdsOperations()
+#        opr.validateApi(api-job)
+# This api can be used to validate the api-job against the
+# perdefined sds_operation_schema in the yaml file.
+
+
+import os
+import six
+import yaml
+import exception as ex
+
+PRIMITIVE_TYPES = {'Boolean': lambda value: isinstance(value, bool),
+                   'Float': lambda value: isinstance(value, float),
+                   'Integer': lambda value: isinstance(value, int),
+                   'Long': lambda value: isinstance(value, (
+                       six.integer_types, float)),
+                   'String': lambda value: isinstance(value, six.string_types),
+                   'Uint': lambda value: isinstance(value, int) and value >= 0,
+                   'List': lambda value: isinstance(value, list)}
+
+
+def loadSchema(schemaFile):
+    try:
+        # Check api operation schema file exists
+        if not os.path.isfile(schemaFile):
+            return (False,
+                    "Error: Schema file '" + schemaFile + "' does not exists.")
+
+        # load api operation schema file and return
+        try:
+            code = open(schemaFile)
+        except IOError, exc:
+            return (False,
+                    "Error loading schema file '" + schemaFile + "': " + exc)
+
+        # Parse schema file
+        try:
+            config = yaml.load(code)
+        except yaml.YAMLError, exc:
+            error_pos = ""
+            if hasattr(exc, 'problem_mark'):
+                error_pos = " at position: (%s:%s)" % (
+                    exc.problem_mark.line+1,
+                    exc.problem_mark.column+1)
+            msg = "Error loading schema file '" + schemaFile + "'" + error_pos \
+                + ": content format error: Failed to parse yaml format"
+            return False, msg
+
+    except Exception, e:
+        return (False, "Error loading api operation schema file '%s': %s" % (
+            schemaFile, str(e)))
+
+    return (True, config)
+
+
+class ApiJobValidator(object):
+    def __init__(self, schemaFilePath, chkMissingYaml=False):
+        self.yamlObj = None
+        self.chkMissingYaml = chkMissingYaml
+        status = loadSchema(schemaFilePath)
+        if not status[0]:
+            raise ex.FailedLoadingSchemaException(1, status[1])
+        yamlConf = status[1]
+
+        # validate yaml schema file
+        if not yamlConf.get('valid_objects'):
+            raise ex.ValidObjectsNotFoundException()
+
+        if not yamlConf.get('object_details'):
+            raise ex.ObjectDetailsNotFoundException()
+
+        if not yamlConf.get('object_details', {}).get("flows"):
+            raise ex.FlowDetailsNotFoundException()
+
+        # load the schema object into memory
+        self.yamlObj = yamlConf.get('object_details')
+
+    def _validateApiJob(self, apiJob):
+        if not isinstance(apiJob, dict) or 'flow' not in apiJob:
```

Comment:
We have changed the api-job structure, please refer https://github.com/Tendrl/bridge_common/blob/master/etc/samples/tendrl_api_job.sample.json

Please update this patch based on the above